src/Numerics.Tests/DistributionTests/Discrete/GeometricTests.cs

```diff
@@ -175,28 +175,21 @@ public void ValidateProbability(double p, int x)
         /// <param name="x">Input X value.</param>
         /// <param name="pln">Expected value.</param>
         [TestCase(0.0, -1, Double.NegativeInfinity)]
-        [TestCase(0.0, 0, 0.0)]
+        [TestCase(0.0, 0, Double.NegativeInfinity)]
         [TestCase(0.0, 1, Double.NegativeInfinity)]
         [TestCase(0.0, 2, Double.NegativeInfinity)]
         [TestCase(0.3, -1, Double.NegativeInfinity)]
-        [TestCase(0.3, 0, -0.35667494393873244235395440410727451457180907089949815)]
+        [TestCase(0.3, 0, Double.NegativeInfinity)]
         [TestCase(0.3, 1, -1.2039728043259360296301803719337238685164245381839102)]
-        [TestCase(0.3, 2, Double.NegativeInfinity)]
+        [TestCase(0.3, 2, -1.5606477482646686)]
         [TestCase(1.0, -1, Double.NegativeInfinity)]
         [TestCase(1.0, 0, Double.NegativeInfinity)]
-        [TestCase(1.0, 1, 0.0)]
+        [TestCase(1.0, 1, Double.NaN)]
         [TestCase(1.0, 2, Double.NegativeInfinity)]
         public void ValidateProbabilityLn(double p, int x, double pln)
```

Comment:
It might be better to have 2 set of tests one using the expected result and the other doing the calculation.

Reply:
It seems the one doing the computation was using exactly the same formula as the implementation, which was kind of pointless. So your changes toward using the provided expected value looks more useful to me.